pyproject.toml

```diff
@@ -67,6 +66,102 @@ mike = "^1.1.2"
+  "B009", # Do not call getattr with a constant attribute value. -- Not always applicable
+  "B010", # Do not call setattr with a constant attribute value. -- Not always applicable
+  "RUF012", # type class attrs with `ClassVar` -- Too strict/trigger-happy
+  "UP007", # Use X | Y for type annotations -- `typer` doesn't support it
+  "COM812", # Checks for the absence of trailing commas -- leads to undesirable behavior from formatters
+  "PIE804", # Unnecessary `dict` kwargs -- Inconvenient to enforce
+  "RET505", # Unnecessary {branch} after return statement -- Lots of false positives
+  "RET506", # Unnecessary {branch} after raise statement -- Lots of false positives
+  "RET507", # Unnecessary {branch} after continue statement -- Lots of false positives
+  "RET508", # Unnecessary {branch} after break statement -- Lots of false positives
+  # "TCH001", # Move application import {} into a type-checking block -- Breaks KPOps
+  "PLR09", # upper bound on number of arguments, functions, etc. -- Inconvenient to enforce
+  "PLR2004", # Magic value used in comparison, consider replacing {value} with a constant variable -- Inconvenient to enforce
+  "PLW2901", # `for` loop variable `{var}` overwritten by assignment target -- Inconvenient to enforce
+  "TRY002", # Create your own exception -- Inconvenient to enforce
+  "TRY003", # Avoid specifying long messages outside the exception class -- Inconvenient to enforce
+]
+select = [
+  "F", # Pyflakes
+  "E", # pycodestyle Errors
+  "W", # pycodestyle Warnings
+  "C90", # mccabe
+  "I", # isort
+  "D", # pydocstyle
+  "UP", # pyupgrade
+  "B", # flake8-bugbear
+  "INP", # flake8-no-pep420
+  "RUF", # Ruff-specific rules
+  "YTT", # flake8-2020
+  "ASYNC", # flake8-async
+  "BLE", # flake8-blind-except
+  "COM", # flake8-commas
+  "C4", # flake8-comprehensions
+  "T10", # flake8-debugger
+  "EM", # flake8-errmsg
+  "FA", # flake8-future-annotations
+  "ISC", # flake8-implicit-str-concat
+  "ICN", # flake8-import-conventions
+  "INP", # flake8-no-pep420
+  "PIE", # flake8-pie
+  "PT", # flake8-pytest-style
+  "Q", # flake8-quotes
+  "RSE", # flake8-raise
+  "RET", # flake8-return
+  "SLOT", # flake8-slots
+  "SIM", # flake8-simplify
+  "TCH", # flake8-type-checking, configure correctly and add
+  "PTH", # flake8-use-pathlib
+  "PGH", # pygrep-hooks
+  "PL", # Pylint
+  "TRY", # tryceratops
+  # "FURB", # refurb, add when out of nursery
+  # "LOG", # flake8-logging, add when out of nursery
+]
+output-format = "grouped"
+show-fixes = true
+task-tags = ["TODO", "HACK", "FIXME", "XXX"]
+target-version = "py310"
+exclude = ["tests/*snapshots/*"]
+
+[tool.ruff.extend-per-file-ignores]
+"tests/*/__init__.py" = ["F401"]
+
+[tool.ruff.isort]
+split-on-trailing-comma = false
+
+[tool.ruff.flake8-bugbear]
+extend-immutable-calls = ["typer.Argument"]
+
+[tool.ruff.flake8-type-checking]
+runtime-evaluated-base-classes = ["pydantic.BaseModel", "kpops.components.base_components.kafka_app.KafkaApp"]
```

Comment:
why KafkaApp?

Reply:
Not sure why that is the fix, but otherwise [TCH001](https://docs.astral.sh/ruff/rules/typing-only-first-party-import/) moves a couple imports into a `if TYPE_CHECKING: ...` block and breaks KPOps. `pydantic.BaseModel` should be enough to prevent that from happening I think, but maybe the `ABC` parent of `KafkaApp` throws `ruff` off.

Reply:
perhaps I am getting a slightly different error, it says to move `StreamsAppConfig` (in `StreamsApp`) and `ProducerValues` (in `StreamsApp`) (*weird naming discrepancy btw) in a type checking block. would that be fine?

Reply:
It's the same error, it caused crashes when I tried, no idea why tbh

Reply:
what do you mean by crashes? kpops runtime errors?

Reply:
yes

Reply:
If you remove `KafkaApp` from the list and let `ruff` autofix the `TCH001` errors with `ruff check . --fix` `pytest` would fail

Reply:
I think we should open an issue for that, as you said flagging `pydantic.BaseModel` should suffice

Reply:
Here is an example: `FAILED tests/components/test_streams_app.py::TestStreamsApp::test_set_topics - pydantic.errors.ConfigError: field "app" not yet prepared so type is still a ForwardRef, you might need to call StreamsApp.update_for
ward_refs().`

Reply:
> I think we should open an issue for that, as you said flagging `pydantic.BaseModel` should suffice. I could imagine that the parent class resolving has a max depth or something similar which causes them not to be detected as children

Could be

Reply:
The problem persists if you replace `KafkaApp` with `PipelineComponent` or `KubernetesApp`

Reply:
Shouldn't be a depth issue in that case I think

Reply:
Also persists with `StreamsApp` in place of `KafkaApp`

Reply:
yep, that's a good title

Reply:
when I remove it the error no longer appears

Reply:
ok, then let's do it like this for now. still sounds like a bug that we should file an issue for, because `StreamsAppConfig` and `ProducerValues` weren't resolved as children of `pydantic.BaseModel`

Reply:
Do you mean that I should get rid of the import?